compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -260,10 +261,10 @@ fn struct_definition() -> impl NoirParser<TopLevelStatement> {
             [(LeftParen, RightParen), (LeftBracket, RightBracket)],
             |_| vec![],
         ))
-        .or(just(Semicolon).map(|_| Vec::new()));
+        .or(just(Semicolon).to(Vec::new()));
 
     attributes()
-        .or_not()
+        .map(Option::Some)
```

Comment:
It looks like `attributes` is always `Some` now, so we don't need the `Option` wrapper after all. Do you mind removing it from where it is used in the Ast (only function definitions IIRC)?